Core/Lib/Models/IMetadataModel.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Lens.Core.Lib.Models;
+
+public interface IMetadataModel
+{
+    public const string AllDomains = "*";
+
+    dynamic? Metadata { get; }
```

Comment:
Why don't you return a JsonNode? Dynamic is a (very) bad code smell and should only be used of the datatype changes (which should never happen in c# imho). This datatype never changes, only the JSON structure, but it remains a JsonNode

Reply:
Same for all other dynamics related to jsonnode

Reply:
@jstorteboom notification!

Reply:
It grew this way historically. Back in the days a viewcontroller had a 'Bag' property which was a dynamic. At this point in time and in this case it is implemented using a JsonNode. This can change of course (though not very likely). So, do we go for the more generic way (dynamic) or more obvious way, since we know we're dealing with json here. I'd say the obvious way and explicitly make it JsonNode.